kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/utils/KubernetesSerialization.java

```diff
@@ -188,6 +188,23 @@ public <T> String asJson(T object) {
    * @return a String containing a JSON representation of the provided object.
    */
   public <T> String asYaml(T object) {
+    return asYaml(object, false);
+  }
+
+  /**
+   * Returns a YAML representation of the given object.
+   *
+   * <p>
+   * If the provided object contains a JsonAnyGetter annotated method with a Map that contains an entry that
+   * overrides a field of the provided object, the Map entry will take precedence upon serialization. Properties won't
+   * be duplicated.
+   *
+   * @param object the object to serialize.
+   * @param minQuotes whether strings will be rendered without quotes (true) or with quotes (false).
+   * @param <T> the type of the object being serialized.
+   * @return a String containing a JSON representation of the provided object.
+   */
+  public <T> String asYaml(T object, boolean minQuotes) {
```

Comment:
I think this change looks good, but wanted to bring up that based upon the DumpSettings there's a lot that could be overriden about the generation style.

Min quoting is roughly defaultScalarStyle=ScalarStyle.PLAIN - it's just that we need to still quote keys that look like boolean values to kube. Another possibliity is to create our own "DumpSettings" builder and have that as an argument so that this can be expanded in the future without adding additional signatures to asYaml.